O/ORTools/build_tarballs.jl

```diff
@@ -0,0 +1,181 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "ORTools"
+version = v"9.7"
+
+# Collection of sources required to build this package
+sources = [
+    GitSource("https://github.com/google/or-tools.git",
+              "6fa02e157a5c91067b7d7b88629472b9ed461193"),
+    DirectorySource("./bundled")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+# Prepare the source directory.
+cd $WORKSPACE/srcdir/or-tools*
+atomic_patch -p1 "${WORKSPACE}/srcdir/patches/cmake_dependencies_CMakeLists.txt.patch"
+mkdir build
+cmake --version
+
+# Make the host compile tools easily accessible when cross-compiling.
+# Otherwise, CMake will use the cross-compiler for host tools.
```

Comment:
I'm very confused by this statement: CMake has no distinction between host and target tools (which is a rather annoying limitation of this build system), we want to compile for the target, not the host, and `CMAKE_TARGET_TOOLCHAIN` should already be setting the right toolchain.

Reply:
And that's exactly the problem: without this, CMake only knows about the target tools and cannot use the host toolchain (which is required for tools that are supposed to run on the host).

Reply:
The usual way to deal with this bootstrap problem is to a first build with the host toolchain (BTW, there's `CMAKE_HOST_TOOLCHAIN` without setting everything manually) to build the needed host tools, and then another one with the target toolchain for building the rest.

Reply:
There's no real bootstrap issue here, as I'm using the existing tool chains (it's just that the build needs a few uncommon tools).

I tried to use the undocumented `CMAKE_HOST_TOOLCHAIN`: unlike the global environment variables, Abseil doesn't detect that this version of GCC defaults to C++17. (At least, the current script builds OR-Tools.)